internal/libyaml/scanner.go

```diff
@@ -1661,20 +1677,23 @@ func (parser *Parser) scanDirective(token *Token) bool {
 
 		// Unknown directive.
 	} else {
-		parser.setScannerError("while scanning a directive",
+		return parser.setScannerError("while scanning a directive",
```

Comment:
I feel like this method should be renamed now.

Something like formatScannerError

Also,maybe there is no need for it to be a method maybe, a function could be enough

Previously it was setting something in the parser struct, now it doesn't any more, right?